src/System.Runtime.Loader/tests/ResourceAssemblyLoadContext.cs

```diff
@@ -43,7 +43,13 @@ protected override Assembly Load(AssemblyName assemblyName)
                 // This custom load context will extract that resource and store it at the %temp% path at runtime.
                 // This prevents the corerun from adding the test assembly to the TPA list.                
                 // Once loaded it is not possible to unload the assembly, therefore it cannot be deleted.
-                string path = Path.Combine(Path.GetTempPath(), assembly);
+                var tempPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+                if (!Directory.Exists(tempPath))
```

Comment:
Do we need to do this check?  If the directory already exists, we have bigger issues (with regards to the uniquess of values produced by NewGuid).